### DISCUSSION TITLE
Throttle the ledger rollover

### DISCUSSION THREAD

**Discussion by codelipenghui** [apache/pulsar#18873] [General]
**Is your feature request related to a problem? Please describe.**
High ledger rollover frequency may lead to zookeeper in heavy workload. In order to make zookeeper run more stable, we should limit the ledger rollover frequency.

**Describe the solution you'd like**
Add a rate limiter for the ledger rollover.

**Comment by Jason918** [comment]
@codelipenghui HI, I'll be happy to participate. 
Please check this more detailed solution.
Add a new parameter `managedLedgerRolloverRateLimit`, rate limit the amount of rollover per second in one broker. This parameter can be set as 0 by default meaning no rate limiting .

As `ManagedLedgerImpl#currentLedgerIsFull` is called before each ledger rollover, we can add a global rate limiter after that. Every ledger rollover can only happen when `currentLedgerIsFull()` and `rolloverRatelimiter.tryAcquire()` both return true.
